edge-hub/src/Microsoft.Azure.Devices.Routing.Core/endpoints/StoringAsyncEndpointExecutor.cs

```diff
@@ -1,143 +1,144 @@
-    {
-        readonly AtomicBoolean closed = new AtomicBoolean();
-        readonly IMessageStore messageStore;
-        readonly Task sendMessageTask;
-        readonly AsyncManualResetEvent hasMessagesInQueue = new AsyncManualResetEvent(true);
-        readonly AsyncEndpointExecutorOptions options;
-        readonly CancellationTokenSource cts = new CancellationTokenSource();
-        readonly ICheckpointerFactory checkpointerFactory;
-        readonly EndpointExecutorConfig config;
-        AtomicReference<SortedDictionary<uint, EndpointExecutorFsm>> prioritiesToFsms;
-        EndpointExecutorFsm lastUsedFsm;
-
-        public StoringAsyncEndpointExecutor(
-            Endpoint endpoint,
-            ICheckpointerFactory checkpointerFactory,
-            EndpointExecutorConfig config,
-            AsyncEndpointExecutorOptions options,
-            IMessageStore messageStore)
-        {
-            this.Endpoint = Preconditions.CheckNotNull(endpoint);
-            this.checkpointerFactory = Preconditions.CheckNotNull(checkpointerFactory);
-            this.config = Preconditions.CheckNotNull(config);
-            this.options = Preconditions.CheckNotNull(options);
-            this.messageStore = messageStore;
-            this.sendMessageTask = Task.Run(this.SendMessagesPump);
-            this.prioritiesToFsms = new AtomicReference<SortedDictionary<uint, EndpointExecutorFsm>>(new SortedDictionary<uint, EndpointExecutorFsm>());
-        }
-
-        public Endpoint Endpoint { get; }
-
-        public EndpointExecutorStatus Status => this.lastUsedFsm.Status;
-
-        public async Task Invoke(IMessage message, uint priority, uint timeToLiveSecs)
-        {
-            try
-            {
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using App.Metrics;
+    using App.Metrics.Counter;
+    using App.Metrics.Timer;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.Concurrency;
+    using Microsoft.Azure.Devices.Routing.Core.Endpoints.StateMachine;
+    using Microsoft.Extensions.Logging;
+    using Nito.AsyncEx;
+    using static System.FormattableString;
+    using AsyncLock = Microsoft.Azure.Devices.Edge.Util.Concurrency.AsyncLock;
+
+    public class StoringAsyncEndpointExecutor : IEndpointExecutor
+    {
+        readonly AtomicBoolean closed = new AtomicBoolean();
+        readonly IMessageStore messageStore;
+        readonly Task sendMessageTask;
+        readonly AsyncManualResetEvent hasMessagesInQueue = new AsyncManualResetEvent(true);
+        readonly AsyncEndpointExecutorOptions options;
+        readonly CancellationTokenSource cts = new CancellationTokenSource();
+        readonly ICheckpointerFactory checkpointerFactory;
+        readonly EndpointExecutorConfig config;
+        AtomicReference<SortedDictionary<uint, EndpointExecutorFsm>> prioritiesToFsms;
+        AtomicReference<EndpointExecutorFsm> lastUsedFsm;

```

Comment:
I suggest we don't bundle such changes together - they are hard to track, revert, etc.